### PR TITLE
add generate_big_data_network script

### PR DESCRIPTION
**Pull request by kstoreyf**
- modified generate_data_network script to generate larger quantities of data (light curves and PSCs) to load test the pipeline, now generate_big_data_network
  - less realistic star data - random coordinates and inaccurate rise and set times
  - more accurate data products - image sequencies, observation of star for each camera
- small updates to original generate_data_network script

these scripts should eventually be combined (they use most of the same code and are currently separated for ease of use case).

### CODE REVIEW

**Comment by kstoreyf** [comment]
can't figure out why this is still failing @wtgee

**Comment by wtgee** [comment]
The `get_site_names` info was removed from astroplan and moved into `astropy.coordinates.EarthLocation`, which is a bit annoying:

```python
from astropy.coordinates import EarthLocation
EarthLocation.get_site_names
```

**Comment by kstoreyf** [comment]
yes i did that (https://github.com/panoptes/PIAA/pull/23/commits/cdc0e874041a7f99fc2e9b0e2b7381582e440135) but it looks like travis doesn't recognize my new commit?

**Comment by wtgee** [comment]
The `get_current_network` still has them. Looks like there are 4 calls total and 2 were changed?

**Comment by kstoreyf** [comment]
wow yep sorry it's late...

File: scripts/generate_big_data_network.py [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+        :param start_date: the start date of observations
+        :param end_date: the end date of observations
+        """
+        start_date = datetime.strptime(start_date, "%Y-%m-%d")
+        end_date = datetime.strptime(end_date, "%Y-%m-%d")
+        num_nights = (end_date - start_date).days + 1
+        if end_date < start_date:
+            raise ValueError('End date must be after start date.')
+
+        units = self.get_current_network()
```
Comment (wtgee) [inline scripts/generate_big_data_network.py]: The `num_units` method that is passed in as a param to the method is ignored. Or at least the usage seems confusing. If I pass `num_units=10` I would expect it generates data for 10 units but it doesn't seem to be doing that.

File: scripts/generate_big_data_network.py [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+                                                 pic, coords, start_time, end_time)
+                            lc = self.build_lightcurve(hdu)
+
+                            # Write data products to local temp files
+                            self.write_psc(
+                                "{}/{}".format(temp_dir, psc_filename), hdu)
+                            self.write_lightcurve(
+                                "{}/{}".format(temp_dir, lc_filename), lc)
+
+                            # Upload data products from local files to cloud
```
Comment (wtgee) [inline scripts/generate_big_data_network.py]: Can we make the upload optional via a command parameter? Is it unrealistic to ever run this on a laptop or personal computer?

File: scripts/generate_big_data_network.py (hunk 1 above)
Comment (kstoreyf) [inline scripts/generate_big_data_network.py]: num_units is used to define the number of units if there is currently no data in the storage bucket (line 64). if there already is a network in the cloud, the units from there are taken and num_units is ignored. it's fine if this functionality changes based on how this should be used, for instance if we want to run it locally (see comment below)

File: scripts/generate_big_data_network.py (hunk 2 above)
Comment (kstoreyf) [inline scripts/generate_big_data_network.py]: not unrealistic, that's what i've been doing - right now it always downloads the files to local temp storage and then uploads them to cloud. though it never clears the temp dir, which should happen once in a while..

it shouldn't be hard to add a command parameter to decide whether to store locally or upload to cloud. only caveat: if just want to upload to cloud without first downloading locally, would need to add a new function to the storage util in POCS, blob's [upload_from_file](https://googlecloudplatform.github.io/gcloud-python/latest/storage-blobs.html) method. currently only uses upload_from_string (which can be used for light curve JSONs but not PSCs) or upload_from_filename. 

i'll quickly add a toggle and option for where to store files locally now.

**Comment by wtgee** [comment]
Looks good. I made it so when not using cloud stuff it doesn't hit the PanStorage. Going to merge.

**Pull request merged**